src/main/java/org/zalando/nakadi/controller/CursorsController.java

```diff
@@ -99,25 +112,31 @@ public CursorsController(final CursorsService cursorsService,
         }
 
         try {
+            final List<NakadiCursor> nakadiCursors = new ArrayList<>();
+            for (SubscriptionCursor c : cursors.getItems()) {
+                nakadiCursors.add(cursorConverter.convert(c));
+            }
             validateSubscriptionReadScopes(client, subscriptionId);
 
             LOG.debug("[COMMIT_CURSORS] scopes validation finished");
 
-            final List<CursorCommitResult> items = cursorsService.commitCursors(streamId, subscriptionId,
-                    cursors.getItems());
+            final List<Boolean> items = cursorsService.commitCursors(streamId, subscriptionId, nakadiCursors);
 
             LOG.debug("[COMMIT_CURSORS] commit finished");
 
-            final boolean allCommited = items.stream()
-                    .allMatch(cursor -> CursorCommitResult.COMMITTED.equals(cursor.getResult()));
-            final ItemsWrapper<CursorCommitResult> body = new ItemsWrapper<>(items);
-            return allCommited ? noContent().build() : ok(body);
-        } catch (final NoSuchEventTypeException e) {
+            final boolean allCommited = items.stream().allMatch(item -> item);
+            if (allCommited) {
+                return noContent().build();
+            } else {
+                final List<CursorCommitResult> body = IntStream.range(0, cursors.getItems().size())
+                        .mapToObj( idx -> new CursorCommitResult(cursors.getItems().get(idx), items.get(idx)))
```

Comment:
You anyway do the same by calling items.get(idx), so the code "safeness" will not change if using solution with iterator, but the code will be more laconic.
Also, you have extra space in that line so you will have to change it anyway :)